modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/servlet/taglib/aui/part/ESMPortletDataPart.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.frontend.js.web.internal.servlet.taglib.aui.part;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.portal.kernel.servlet.taglib.aui.AMDRequire;
+import com.liferay.portal.kernel.servlet.taglib.aui.ESMImport;
+import com.liferay.portal.kernel.servlet.taglib.aui.PortletDataPart;
+import com.liferay.portal.kernel.servlet.taglib.aui.VariableUtil;
+import com.liferay.portal.kernel.util.StringBundler;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author Iván Zaera Avellón
+ */
+public class ESMPortletDataPart implements PortletDataPart {
+
+	public ESMPortletDataPart(
+		String content, String amdRequire, Collection<ESMImport> esmImports) {
+
+		_content = StringUtil.trim(content);
+		_esmImports = esmImports;
+
+		Map<String, String> moduleAliasMap = VariableUtil.parseRequire(
+			amdRequire);
+
+		for (Map.Entry<String, String> entry : moduleAliasMap.entrySet()) {
+			_amdRequires.add(new AMDRequire(entry.getKey(), entry.getValue()));
+		}
+	}
+
+	@Override
+	public Collection<AMDRequire> getAMDRequires() {
+		return _amdRequires;
+	}
+
+	@Override
+	public List<String> getAUIModules() {
+		return null;
+	}
+
+	@Override
+	public String getContent(
+		Map<AMDRequire, AMDRequire> amdRequireMap,
+		Map<ESMImport, ESMImport> esmImportMap) {
+
+		if (Validator.isNull(_content) && _amdRequires.isEmpty() &&
+			_esmImports.isEmpty()) {
+
+			return StringPool.BLANK;
+		}
+
+		StringBundler sb = new StringBundler();
+
+		sb.append("{\n");
+
+		for (ESMImport esmImport : _esmImports) {
+			String alias = esmImport.getAlias();
+
+			ESMImport mappedESMImport = esmImportMap.get(esmImport);
+
+			String variable = mappedESMImport.getAlias();
+
+			if (!alias.equals(variable)) {
+				sb.append("const ");
```

Comment:
You use `var` in some places and `const` in others, is there a particular reason for that?

Reply:
I use `var` for the "legacy" code (AUI and AMD) because we used `var` before. On the opposite, I use `const` for the new ESM code so that we make it safer.